framework/behaviors/SluggableBehavior.php

```diff
@@ -142,9 +148,12 @@ protected function getValue($event)
             if ($this->isNewSlugNeeded()) {
                 $slugParts = [];
                 foreach ((array) $this->attribute as $attribute) {
-                    $slugParts[] = ArrayHelper::getValue($this->owner, $attribute);
+                    $part = ArrayHelper::getValue($this->owner, $attribute);
+                    if ($this->skipOnEmpty && $this->isEmpty($part)) {
+                        return $this->owner->{$this->slugAttribute};
```

Comment:
I am unsure if this logic is correct:
- if there is no slug (e.g. for new record) and its source is empty - the slug will be empty.
- if slug is set (existing record) and its source is empty (but changed!) - the slug remains unchanged, which seems odd